autoblocks/_impl/tracer.py

```diff
@@ -264,16 +290,38 @@ def send_event(
         the event failed, the trace_id will be None.
         """
         try:
+            merged_properties = dict(self._properties)
```

Comment:
Would be nice to pull all of this property merging stuff into a function just bc I think functions with a large amount of lines in a try/except are hard to follow